codegen/facelift/templates/ServiceIPC.template.h

```diff
@@ -367,38 +389,45 @@ class {{classExport}} {{className}} : public facelift::IPCProxy<{{interfaceName}
 
     void deserializeSignal(facelift::IPCMessage &msg) override
     {
-        QString signalName;
-        deserializeValue(msg, signalName);
+        SignalID member;
+        deserializeValue(msg, member);
 
+        switch (member) {
         {% for event in interface.signals %}
-        if (signalName == "{{event}}") {
+        case SignalID::{{event}}: {
             {% for parameter in event.parameters %}
             {{parameter.interfaceCppType}} param_{{parameter.name}};
             deserializeValue(msg, param_{{parameter.name}});
             {% endfor %}
-            {{event}}(
+            emit {{event}}(
             {%- set comma = joiner(", ") -%}
             {%- for parameter in event.parameters -%}
                 {{ comma() }}param_{{parameter.name}}
             {%- endfor -%}  );
-        }
+        } break;
```

Comment:
maybe align the 'break' with code above (i.e. three spaces between '}' and 'break'). Looks nicer and you might consider removing the extra scope where it is not needed.